clients/ui/frontend/src/app/utils/useFetch.ts

```diff
@@ -0,0 +1,278 @@
+import * as React from 'react';
```

Comment:
we have this hook `useFetchState` in shared library, I think we can import it from there.